pkgs/by-name/si/sillytavern/package.nix

```
@@ -0,0 +1,41 @@
+{
+  buildNpmPackage,
+  nodejs,
+  fetchFromGitHub,
+  stdenv,
+  lib,
+}:
+buildNpmPackage rec {
+  pname = "sillytavern";
+  version = "1.12.10";
+  src = fetchFromGitHub {
+    owner = "SillyTavern";
+    repo = "SillyTavern";
+    tag = version;
+    sha256 = "sha256-eCGDqG0dLCkOPfSunzdskE9PbmXnii96X10KwIMUaeY=";
+  };
+  npmDepsHash = "sha256-KwPzpw85bCBfobARIlyaX4r0rRh3SlaGE7rNlPEvD6A=";
+  dontNpmBuild = true;
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/{bin,app}
+    cp -r . $out/app/sillytavern
+    makeWrapper ${nodejs}/bin/node $out/bin/sillytavern --add-flags $out/app/sillytavern/server.js
```

```suggestion
    makeWrapper ${nodejs}/bin/node $out/bin/sillytavern \
      --add-flags $out/app/sillytavern/server.js \
      --set NODE_ENV production
```
It's better to just [keep up with upstream](https://github.com/SillyTavern/SillyTavern/blob/d0e1c3287a828eff65a2c92380aaad9c38531cee/start.sh#L28) and set NODE_ENV although it's not doing much at the moment.